stbt-batch.d/run.py

```diff
@@ -152,6 +150,119 @@ def on_term(_signo, _frame):
+            return subprocess.check_output(('git',) + cmd, cwd=testdir).strip()
+        return GitInfo(
+            git('describe', '--always', '--dirty'),
+            git('rev-parse', 'HEAD'),
+            git('rev-parse', '--show-toplevel'))
+    except subprocess.CalledProcessError:
+        return None
+    except OSError as e:
+        # ENOENT means that git is not in $PATH
+        if e.errno != errno.ENOENT:
+            raise
+
+
+def make_rundir(outputdir, tag):
+    for n in range(2):
+        rundir = datetime.datetime.now().strftime("%Y-%m-%d_%H.%M.%S") + tag
+        try:
+            os.mkdir(os.path.join(outputdir, rundir))
+            return rundir
+        except OSError as e:
+            if e.errno == errno.EEXIST and n == 0:
+                # Avoid directory name clashes if the previous test took <1s to
+                # run
+                time.sleep(1)
+            else:
+                raise
+
+
+def fill_in_data_files(rundir, test_name, test_args, git_info, tag):
+    def write_file(name, data):
+        with open(os.path.join(rundir, name), 'w') as f:
+            f.write(data)
+
+    if git_info:
+        write_file("git-commit", git_info.commit)
+        write_file("git-commit-sha", git_info.commit_sha)
+        write_file("test-name", os.path.relpath(test_name, git_info.git_dir))
+    else:
+        write_file("test-name", test_name)
+    write_file("test-args", "\n".join(test_args))
+
+    if tag:
+        write_file("extra-columns", "Tag\t%s\n" % tag)
+
+
+@contextmanager
+def setup_dirs(outputdir, tag, state_sender):
+    mkdir_p(outputdir)
+
+    rundir = make_rundir(outputdir, tag)
+
+    symlink_f(rundir, os.path.join(outputdir, "current" + tag))
+
+    state_sender.set({
+        "active_results_directory":
+        os.path.abspath(os.path.join(outputdir, rundir))})
+    try:
+        yield os.path.join(outputdir, rundir)
+    finally:
+        # Now test has finished...
+        symlink_f(rundir, os.path.join(outputdir, "latest" + tag))
+        state_sender.set({"active_results_directory": None})
+
+
+def symlink_f(source, link_name):
+    name = "%s-%06i~" % (link_name, random.randint(0, 999999))
+    os.symlink(source, name)
+    os.rename(name, link_name)
+
+
+DEVNULL_R = open('/dev/null')
+
+
+def run_one(test_name, test_args, batch_args, cwd):
+    """
+    Invoke the run-one shell-script with the appropriate arguments.
+    """
+    test_file = os.path.abspath(test_name)
+
+    cmd = [_find_file('../stbt-run'), '--save-thumbnail=always']
+    if batch_args.do_save_video:
+        cmd += ['--save-video=video.webm']
+    if batch_args.debug:
+        cmd += ['-vv']
+    else:
+        cmd += ['-v']
+    cmd += [test_file, '--'] + list(test_args)
+
+    subenv = dict(os.environ)
+    subenv['do_html_report'] = "true" if batch_args.do_html_report else "false"
+    subenv['stbt_root'] = _find_file('..')
+    subenv['test_displayname'] = " ".join(
+        (os.path.basename(test_name),) + test_args)
```

Comment:
This is a regression in behaviour (from master, not from the previous commit). We should display the test file as given on the command line -- not the basename, and not the abspath either. I'll push a fixup.